### DISCUSSION TITLE
Migrating a project from InstallShield: Is there a tool, or at least a guide?

### DISCUSSION THREAD

**Discussion by GuyFleegman** [wixtoolset/issues#7171] [Questions]
I have an existing project (solution) in Visual Studio 2012, and currently it uses InstallShield - via its VS integration - to create MSI packages.
I hate InstallShield and want to get away from it.  It seems like WiX is the way to go.
But support does WiX have for migrating my existing IS project?  Any tools, or at least a guide?
Thanks!

**Comment by robmen** [comment]
The WiX Toolset provides a "decompiler" that converts an MSI (or MSM) into .wxs source code. In WiX v4 use the `wix msi decompile` command.

Once you have the .wxs code, you can start cleaning it up since it was machine generated.

**Reply by umerchaudhary** [reply]
hello
I have installed Wix toolset v3. I added environment variable "C:\Program Files (x86)\WiX Toolset v3.11\bin" in User Variles/Path
Still i am not able to use wix commands in cmd. Its still giving error "'wix' is not recognized as an internal or external command,
operable program or batch file."

**Reply by BMurri** [reply]
WiX v4 (released this year) has the `wix` tool. WiX v3 (now obsolete) had a series of tool names (none of which was "wix").

The docs for [v3](https://wixtoolset.org/docs/v3/) and for [v4](https://wixtoolset.org/docs/intro/).

I'd highly recommend v4.
